src/space_usage.rs

```diff
@@ -0,0 +1,123 @@
+//! Module to collect and display Disk Space Usage of a Profile.
+use crate::context::Context;
+use anyhow::Result;
+use humansize::{BINARY, format_size};
+
+/// Space Usage Report
+/// Useful for debugging space usage problems in the deltachat database.
+#[derive(Debug)]
+pub struct SpaceUsage {
+    /// Total database size, subtract this from the backup size to estimate size of all blobs
+    pub db_size: usize,
+    /// size and row count of the 10 biggest tables
+    pub largest_tables: Vec<(String, usize, Option<usize>)>,
+    /// count and total size of status updates
+    /// for the 10 webxdc apps with the most size usage in status updates
+    pub largest_webxdc_data: Vec<(usize, usize, usize)>,
+}
+
+impl std::fmt::Display for SpaceUsage {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let mut report = String::from("SpaceUsage:\n");
+        let human_db_size = format_size(self.db_size, BINARY);
+        report += &format!("[Database Size]: {human_db_size}\n");
+        report += "[Largest Tables]:\n";
+        for (name, size, row_count) in &self.largest_tables {
+            let human_table_size = format_size(*size, BINARY);
+            report += &format!(
+                "   {name:<20} {human_table_size:>10}, {row_count:>6} rows\n",
+                name = format!("{name}:"),
+                row_count = row_count.map(|c| c.to_string()).unwrap_or("?".to_owned())
+            );
+        }
+        report += "[Webxdc With Biggest Status Update Space Usage]:\n";
+        for (msg_id, size, update_count) in &self.largest_webxdc_data {
+            let human_size = format_size(*size, BINARY);
+            report += &format!(
+                "   {msg_id:<8} {human_size:>10} across {update_count:>5} updates\n",
+                msg_id = format!("{msg_id}:")
+            );
+        }
+        write!(f, "{report}")
+    }
+}
+
+impl Context {
+    /// Get space usage information for the Context's database
+    /// used in Context.get_info()
+    pub async fn get_space_usage(&self) -> Result<SpaceUsage> {
```

Comment:
I think it is better to have a free function `get_space_usage(context: &Context)` without additional indentation.

I know that we already stuff a lot of functions into `Context`, but I think we should not do more of this. E.g. for `Context.get_webxdc_status_updates()`, it is not found in https://rs.delta.chat/deltachat/webxdc/index.html but on a huge page for `Context` here which is full of `impl Context` sections: https://rs.delta.chat/deltachat/context/struct.Context.html#method.get_webxdc_status_updates